openapi/controllers/main.py

```diff
@@ -39,3 +49,77 @@ def copy(obj, **d):
         }})
 
         return result
+
+
+class OAS(http.Controller):
+
+    @http.route('/api/v1/<namespace_name>/swagger.json',
+                type='http', auth='public', csrf=False)
+    def OAS_json_spec_download(self, namespace_name, **kwargs):
+        namespace = http.request.env['openapi.namespace'].sudo().search([('name', '=', namespace_name)])
+        if not namespace:
+            raise werkzeug.exceptions.NotFound()
+
+        current_host = http.request.env['ir.config_parameter'].sudo().get_param('web.base.url')
+        parsed_current_host = urlparse.urlparse(current_host)
+
+        spec = {
+            "swagger": "2.0",
+            "info": {
+                "title": "Swagger Sample App",
+                "version": "1.0.0"
+            },
+            "host": parsed_current_host.netloc,
+            "basePath": "/v1/%s" % namespace_name,
+            "schemes": [
+                "https",
+                "http"
+            ],
+        }
+
+        try:
+            spec.update(namespace.get_OAS_part())
+        except SwaggerValidationError as e:
+            _logger.exception('namespace.get_OAS_part')
+            return "OAS scheme generating error: %s" % (str(e) or repr(e))
+
+        sorted_spec = OrderedDict([
+            (property_name, spec[property_name])
+            for property_name in SWAGGER_ORDERED_MAIN_PROPERTIES
+            if spec.get(property_name)
+        ])
+
+        response_params = {
+            'headers': [('Content-Type', 'application/json')]
+        }
+        if 'download' in kwargs:
+            response_params['headers'] = [
+                ('Content-Type', 'application/octet-stream; charset=binary'),
+                ('Content-Disposition', content_disposition('swagger.json')),
+            ]
+            response_params['direct_passthrough'] = True
+
+        return werkzeug.wrappers.Response(
+            json.dumps(sorted_spec),
+            status=200,
+            **response_params
+        )
+
+
+SWAGGER_ORDERED_MAIN_PROPERTIES = [
```

Comment:
Module globals  are usually at a beginning of a module, after imports
why you use global var for this?